app/controllers/films_controller.rb

```diff
@@ -11,6 +11,10 @@ def show
       format.html
       format.js
     end
+
+    @film_relateds = Film.related_films @film.category_ids
```

Comment:
Film.related(@film).order_desc
scope :related, ->(film) do
joins(:film_categories).where(category_id: film.category_ids).not(film_id: film.id).distinct.limit 3
end